### PR TITLE
fix accessibility insights reports

### PR DESCRIPTION
**Pull request by olamothe**
These changes are aimed to fix issues reported by this tool: https://accessibilityinsights.io/

image

The Chrome lighthouse report tool did not pick those up (score was 100%, still is 100% after this).

I also ran the local tests with axe-core and everything seems good... 🤷 

* Moved `navigation` role to the Pager itself, instead of the list inside the pager
* Made the links in pager `role="listitem"`, since the parent is now a list instead of `navigation`
* In results per page, removed `-1` tabindex and changed to a simple span, so that it's no longer detected as interactive

https://coveord.atlassian.net/browse/JSUI-3382

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/ResultsPerPage/ResultsPerPage.ts [hunk 1]
```diff
@@ -254,10 +254,9 @@ export class ResultsPerPage extends Component {
 
       listItem.appendChild(
         $$(
-          'a',
+          'span',
```
Comment (ThibodeauJF) [inline src/ui/ResultsPerPage/ResultsPerPage.ts:257]: ❗❗❗ 

Why the "breaking" DOM change?

File: src/ui/Pager/Pager.ts [hunk 2]
```diff
@@ -140,11 +140,15 @@ export class Pager extends Component {
       this.handleQueryStateNumberOfResultsPerPageChanged(data)
     );
     this.addAlwaysActiveListeners();
+    if (!this.element.getAttribute('role')) {
+      this.element.setAttribute('role', 'navigation');
+    }
+    if (!this.element.hasAttribute('aria-label')) {
+      this.element.setAttribute('aria-label', l('Pagination'));
+    }
```
Comment (ThibodeauJF) [inline src/ui/Pager/Pager.ts:148]: I'd move this into a `this.addAccessibilityAttributes();` method

**Review by ThibodeauJF — APPROVED** [review]

File: src/ui/Pager/Pager.ts [hunk 3]
```diff
@@ -432,6 +438,7 @@ export class Pager extends Component {
       .withElement(nextButton)
       .withLabel(l('Next'))
       .withSelectAction(() => this.handleClickNext())
+      .withRole('listitem')
```
Comment (btaillon) [inline src/ui/Pager/Pager.ts:441]: The `listitem` role doesn't imply that this element is selectable, and the `span` a child doesn't either. We should have a selectable role somewhere.

also, IMO, the error reported by the automated tool seems pretty irrelevant.

**Review by btaillon — CHANGES_REQUESTED** [review]
I don't think we should do this
